File: Week 1/Sample Codes/A_oddgnome.py

def find_missing_gnome(gnomes): 
    n = len(gnomes)
    for i in range(1, n - 1):
        sorted = True
        prev = -1
        for j in range(0, n):
            if i == j:
                continue
            sorted = sorted and gnomes[j] >= prev
            prev = gnomes[j]

        if sorted:
            return i


trials = int(input())
for trial in range(trials):
    gnomes = list(map(int, input().split()))
    n = gnomes[0]   
    gnomes = gnomes[1:]
    print(  find_missing_gnome(gnomes) )

File: Week 1/Sample Codes/A_oddgnome.cpp
#include <bits/stdc++.h>
using namespace std;


int main(){
    int trials;
    cin >> trials;
    for(int t = 0; t < trials; t++){
        int n;
        cin >> n;
        vector<int> gnomes(n);
        for(int i = 0; i < n; i++){
            cin >> gnomes[i];
        }
        for(int i = 1; i < n - 1; i++){ //i = candidate for the king
            int prev = -1;
            bool sorted = true;
            for(int j = 0; j < n; j++){
                if(i == j) continue;
                if(gnomes[j] < prev){
                    sorted = false;
                }
                prev = gnomes[j];
            }
            if(sorted){
                cout << i << endl;
            }
        }
    }
}   